app/main.py

```diff
@@ -1,25 +1,99 @@
+from __future__ import annotations
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int
+    ) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(
+            self,
+            owner: SlideLimitationValidator,
+            name: str
+    ) -> None:
```

Comment:
You have a type hinting issue here. The `owner` argument in the `__set_name__` method is expected to be of `SlideLimitationValidator` type, but you are passing `self` which is of `IntegerRange` type. Consider reviewing your type hints.